openslides/agenda/static/js/agenda/projector.js

```diff
@@ -76,7 +76,44 @@ angular.module('OpenSlidesApp.agenda.projector', ['OpenSlidesApp.agenda'])
         // Add it to the coresponding get_requirements method of the ProjectorElement
         // class.
 
+        // Converts the array of agendaitems into a real tree structure.
+        // Every node contains an array of children and the item. The return
+        // value is a simple array of nodes representing the top layer of nodes.
+        // requires an flat tree of agendaitems with a 'parentCount' that
+        // gives the number of parents above the item.
+        var getRealTree = function (items) {
+            var startNodes = [];
+            for (var i = 0; i < items.length; i++) {
```

Comment:
Hm. This code looks slow: The loop runs over all items, also for the children of the children. The problem is that the lopp in line 105 also runs through all items again and again ...

Is this a performance issue or do you think it is ok even for a deeply nested agenda with 500 entries (one for each motion e. g.)? Maybe just test with 1000 somehow nested motions ...

Reply:
You are right; I didn't see that we already implemented this sort of method: The `AgendaTree` has a method `getTree`. I'll use this.